api/RS.API/RS.Service/Logic/CandidateManagerService.cs

```diff
@@ -4,19 +4,171 @@
 using RS.ViewModel.User;
 using System;
 using RS.Data.Interfaces;
+using RS.Common.CommonData;
+using RS.ViewModel.Candidate;
+using RS.Entity.Models;
+using System.Security.Claims;
+using System.Security.Principal;
+using RS.Common.Extensions;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace RS.Service.Logic
 {
     public class CandidateManagerService : ICandidateManagerService
     {
+        private readonly ClaimsPrincipal _principal;
         #region Global Variables
         private readonly ICandidateRepository _candidateRepository;
+        private readonly IOpeningRepository _openingRepository;
+        private readonly IQualificationRepository _qualificationRepository;
         #endregion
-        public CandidateManagerService(ICandidateRepository candidateRepository)
+        public CandidateManagerService(IPrincipal principal, ICandidateRepository candidateRepository, IOpeningRepository openingRepository, IQualificationRepository qualificationRepository)
         {
+            _principal = principal as ClaimsPrincipal;
             _candidateRepository = candidateRepository;
+            _openingRepository = openingRepository;
+            _qualificationRepository = qualificationRepository;
         }
 
+        public IResult AddCandidate(CandidateViewModel candidate)
+        {
+            var result = new Result
+            {
+                Operation = Operation.Create,
+                Status = Status.Success
+            };
+            try
+            {
+                var candidateModel = new Candidates();
+                candidateModel.MapFromViewModel(candidate, (ClaimsIdentity)_principal.Identity);
+                candidateModel.QualificationId = candidate.Qualification;
+                candidateModel.Qualification = _qualificationRepository.GetByID(candidate.Qualification);
+
+                var openingCandidate = new OpeningCandidates();
+                openingCandidate.OpeningId = candidate.Opening;
+                openingCandidate.candidate = candidateModel;
+                openingCandidate.Opening = _openingRepository.GetByID(candidate.Opening);
+                openingCandidate.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+
+                var organization = new Organizations();
+                organization.Name = candidate.Organization;
+                organization.MapAuditColumns((ClaimsIdentity)_principal.Identity);
+
+                _candidateRepository.AddCandidate(candidateModel, openingCandidate, organization);
+                _candidateRepository.SaveChanges();
```

Comment:
why again save changes used here ? if every thing is in repository layer

Reply:
Done